File: _designs_001/LRU Cache/Solution.java
public class Solution {
    public List<Integer> testLRUCache(StreamTokenizer tokenizer, List<Double> metrics) throws IOException {
        LRUCache cache = new LRUCache(Serializer.deserializeInt(tokenizer));
        List<Integer> result = new ArrayList<Integer>();

        int opsCnt = Serializer.deserializeInt(tokenizer);
        for (int i = 0; i < opsCnt; i++) {
            int m, n;
            double startTime;
            switch (Serializer.deserializeInt(tokenizer)) {
            case 0:
                m = Serializer.deserializeInt(tokenizer);
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                cache.set(m, n);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 1:
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                result.add(cache.get(n));
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            }
        }
        return result;
    }
}

File: _designs_001/LRU Cache/Solution.rb
class Solution
  def testLRUCache(inp, metrics)
    metrics[0] = 0
    lru, r = LRUCache.new(Serializer.deserializeInt(inp)), []
    Serializer.deserializeInt(inp).times do
      case Serializer.deserializeInt(inp)
      when 0
        m, n = Serializer.deserializeInt(inp), Serializer.deserializeInt(inp)
        start_time = Time.now
        lru.set(m, n)
        metrics[0] += (Time.now - start_time) * 1000
      when 1
        n = Serializer.deserializeInt(inp)
        start_time = Time.now
        r << lru.get(n)
        metrics[0] += (Time.now - start_time) * 1000
      end
    end
    r
  end
end